felix.go

```diff
@@ -269,24 +268,46 @@ configRetry:
 	asyncCalcGraph := calc.NewAsyncCalcGraph(configParams, dpConnector.ToDataplane)
 
 	if configParams.UsageReportingEnabled {
-		// Usage reporting enabled, add stats collector to graph and
-		// start the usage reporting thread.
-		statsChan := make(chan calc.StatsUpdate, 1)
+		// Usage reporting enabled, add stats collector to graph.  When it detects an update
+		// to the stats, it makes a callback, which we use to send an update on a channel.
+		// We use a buffered channel here to avoid blocking the calculation graph.
+		statsChanIn := make(chan calc.StatsUpdate, 1)
```

Comment:
Fair enough.  Happy to check my understanding, but also happy to approve the change as is.